### PR TITLE
Perplexity: add clipping and from_logits

### PR DESCRIPTION
**Pull request by jeffcarp**
It was pointed out that Perplexity returns NaNs for negative values. This is because our implementation did not clip logit values to [0, 1], whereas the Keras implementation does. [1]

Even with that fix, the tests were failing because Keras defaults to the TensorFlow version of the metric, which applies softmax to the outputs unconditionally [2], unlike the JAX implementation which does not. [3]

Also:

- Added a `from_logits` arg, similar to Keras, for users who want to pass raw logits and have us apply softmax internally.
- Forced all Keras metrics in tests to use the JAX backend for parity.

[1] https://github.com/keras-team/keras/blob/3f8b065e82b17884bd43fcfbd4bd79f18a7019fe/keras/src/backend/jax/nn.py#L582
[2] https://www.tensorflow.org/api_docs/python/tf/nn/sparse_softmax_cross_entropy_with_logits
[3] https://github.com/keras-team/keras/blob/3f8b065e82b17884bd43fcfbd4bd79f18a7019fe/keras/src/backend/jax/nn.py#L578-L579

### CODE REVIEW

**Comment by jeffcarp** [comment]
Looking into the test failures...

**Comment by jeffcarp** [comment]
Found the issue - when Keras is imported in other test files first it doesn't have `KERAS_BACKEND` set correctly.
